module/os/linux/spl/spl-kmem-cache.c

```diff
@@ -177,12 +166,7 @@ kv_alloc(spl_kmem_cache_t *skc, int size, int flags)
 	gfp_t lflags = kmem_flags_convert(flags);
 	void *ptr;
 
-	if (skc->skc_flags & KMC_KMEM) {
-		ASSERT(ISP2(size));
-		ptr = (void *)__get_free_pages(lflags, get_order(size));
-	} else {
-		ptr = spl_vmalloc(size, lflags | __GFP_HIGHMEM);
-	}
+	ptr = spl_vmalloc(size, lflags | __GFP_HIGHMEM);
```

Comment:
This changes the behavior so it always perform a `vmalloc` instead of either performing (effectively) a `kmalloc` or `vmalloc`.   You want to use `spl_kvmalloc` here so the behavior is changed to always prefer a `kmalloc` but if that fails it's ok to fallback to a less efficient `vmalloc`.